list/sort/sort.js

```diff
@@ -91,33 +91,43 @@ steal('can/util', 'can/list', function () {
 			return a === b ? 0 : a < b ? -1 : 1;
 		},
 		_changes: function (ev, attr, how, newVal, oldVal) {
+			var dotIndex = ("" + attr).indexOf('.');
 
 			// If a comparator is defined and the change was to a
 			// list item, consider moving the item.
-			if (this.comparator && /^\d+/.test(attr)) {
-	
-				if (ev.batchNum && ev.batchNum !== this._lastBatchNum) {
-					this.sort();
-					this._lastBatchNum = ev.batchNum;
-					return;
+			if (this.comparator && dotIndex) {
+
+				if (ev.batchNum) {
+					if (ev.batchNum === this._lastProcessedBatchNum) {
+						return;
+					} else {
+						this.sort();
+						this._lastProcessedBatchNum = ev.batchNum;
+						return;
+					}
```

Comment:
My previous batch handling was joke, which inspired the `can.batch.debounce` discussion here: https://github.com/bitovi/canjs/issues/1708